js/modules/resolution.go

```diff
@@ -229,6 +238,10 @@ func (mr *ModuleResolver) LoadMainModule(pwd *url.URL, specifier string, data []
 		panic("somehow running source data for " + specifier + " didn't produce a cyclic module record")
 	}
 
+	unknownModules := mr.unknownModules()
```

Comment:
```suggestion
	if mr.unknownModules() > 0 {
		return newUnknownModulesError(mr.unknownModules())
	}
```
We can probably avoid to allocate it for the happy path